File: app/src/main/java/com/hkb48/keepdo/settings/DoneIconPreference.java
package com.hkb48.keepdo.settings;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;

import androidx.annotation.Nullable;
import androidx.preference.DialogPreference;

class DoneIconPreference extends DialogPreference {

    public DoneIconPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected Object onGetDefaultValue(TypedArray a, int index) {
        return a.getString(index);
    }

    @Override
    protected void onSetInitialValue(@Nullable Object defaultValue) {
        persistString((String) defaultValue);
    }

    public void updatePreference(String newValue) {
        if (callChangeListener(newValue)) {
            persistString(newValue);
        }
    }
}

File: app/src/main/java/com/hkb48/keepdo/settings/DoneIconPreference.kt
package com.hkb48.keepdo.settings

import android.content.Context
import android.content.res.TypedArray
import android.util.AttributeSet
import androidx.preference.DialogPreference

internal class DoneIconPreference(context: Context?, attrs: AttributeSet?) :
    DialogPreference(context, attrs) {
    override fun onGetDefaultValue(a: TypedArray, index: Int): Any {
        return a.getString(index)!!
    }

    override fun onSetInitialValue(defaultValue: Any?) {
        persistString(defaultValue as String?)
    }

    fun updatePreference(newValue: String?) {
        if (callChangeListener(newValue)) {
            persistString(newValue)
        }
    }
}